src/java.base/share/classes/java/lang/String.java

```diff
@@ -4270,6 +4270,19 @@ public String translateEscapes() {
                 case 't':
                     ch = '\t';
                     break;
+                case 'u':
+                    if (from + 4 <= length) {
+                        String hex = substring(from, from + 4);
+                        from += 4;
+                        try {
+                            ch = (char) Integer.parseInt(hex, 16);
+                        } catch (NumberFormatException ex) {
+                            throw new IllegalArgumentException("Invalid unicode sequence: " + hex);
+                        }
```

Comment:
Avoids an allocation on a valid sequence, but is perhaps slower.
```suggestion
                        from += 4;
                        try {
                            ch = (char) Integer.parseInt(this, from - 4, from, 16);
                        } catch (NumberFormatException ex) {
                            throw new IllegalArgumentException("Invalid unicode sequence: " + substring(from - 4, from));
                        }
```

Reply:
Good.